### PR TITLE
Revamp checkbox form field

### PR DESCRIPTION
**Pull request by rcorral**
* Moved existing `FieldCheckbox` to `FieldCheckboxMultiple`
* Created a `FieldCheckbox` which will only render a single checkbox.
* Added disabled state for checkboxes.

TODO:
* `FieldCheckboxMultiple` still doesn't really work the way it probably should. It should probably only take in one name and use that for all checkboxes like so `name="foo[]"`
* Still no "edit" state

img

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/Form/__tests__/FieldCheckbox-test.js [hunk 1]
```diff
@@ -11,35 +11,32 @@ describe('FieldCheckbox', function () {
 
   describe('#hasError', function () {
     it('should return true when error contains name', function () {
-      var instance = TestUtils.renderIntoDocument(
+      var instance = instance = TestUtils.renderIntoDocument(
```
Comment (kennyt) [inline src/Form/__tests__/FieldCheckbox-test.js]: i'm seeing double

File: src/Form/FieldInput.js [hunk 2]
```diff
@@ -202,7 +202,7 @@ FieldInput.propTypes = {
     React.PropTypes.string,
     React.PropTypes.bool
   ]),
-  // initial value of checkbox, should be either 'checked' or 'unchecked'
+  // initial value of field, should be either 'checked' or 'unchecked'
```
Comment (kennyt) [inline src/Form/FieldInput.js]: why are the docs of `FieldInput` talking about checked/unchecked?

Comment (rcorral) [inline src/Form/FieldInput.js]: Nooo idea lol. Removed.

Comment (rcorral) [inline src/Form/FieldInput.js]: @mlunoe with copy/pasta fun
https://github.com/mesosphere/reactjs-components/commit/fcb005ca09d59f0bb53a89f0e63da0630af93d73#diff-56a980b3f33a388ae085cd222f0fa748R155

File: src/Form/__tests__/FieldCheckbox-test.js (hunk 1 above)
Comment (rcorral) [inline src/Form/__tests__/FieldCheckbox-test.js]: Fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+## Uversioned
```
Comment (daiyiwhale) [inline CHANGELOG.md]: Unversioned?

**Comment by daiyiwhale** [comment]
:+1:

File: src/Form/ItemCheckbox.js [hunk 4]
```diff
@@ -96,6 +101,11 @@ ItemCheckbox.propTypes = {
...
   // should be either true for 'checked' or false 'unchecked'
   checked: React.PropTypes.bool,
+  // Optional value for setting disabled state
+  disabled: React.PropTypes.bool,
+  // Optional. Which field property is currently being edited
+  // (usually passed down from form definition)
+  editing: React.PropTypes.string,
   // Function to handle change event
   // (usually passed down from FieldCheckbox)
   handleChange: React.PropTypes.func,
```
Comment (daiyiwhale) [inline src/Form/ItemCheckbox.js]: Oops I just realize this prop hasn't been renamed to `handleEvent` :'D

Comment (daiyiwhale) [inline src/Form/ItemCheckbox.js]: Oh wait nevermind, I suppose the event handler prop name is only standardized among the FieldTypes

**Pull request merged**